src/sagemaker/session.py

```diff
@@ -127,6 +131,8 @@ def _initialize(self, boto_session, sagemaker_client, sagemaker_runtime_client):
 
         prepend_user_agent(self.sagemaker_runtime_client)
 
+        self.sts_endpoint_url = sts_endpoint_url
```

Comment:
You are setting the sts endpoint url twice: one in the constructor and one in this private method.